File: C/test/test_findPermutationDifference.c

#ifdef TEST

#include "unity.h"

#include "findPermutationDifference.h"

void setUp(void)
{
}

void tearDown(void)
{
}

void test_findPermutationDifference_Example1(void)
{
	char *s = "abc", *t = "bac";
	int expect = 2;
	TEST_ASSERT_EQUAL(expect, findPermutationDifference(s, t));
}

void test_findPermutationDifference_Example2(void)
{
	char *s = "abcde", *t = "edbac";
	int expect = 12;
	TEST_ASSERT_EQUAL(expect, findPermutationDifference(s, t));
}

#endif // TEST

File: Rust/tests/test_find_permutation_difference.rs
use leetcode_rust::solution::Solution;

#[test]
fn test_example_1() {
    let input0 = "abc".to_string();
    let input1 = "bac".to_string();
    let expected = 2;
    assert_eq!(Solution::find_permutation_difference(input0, input1), expected);
}

#[test]
fn test_example_2() {
    let input0 = "abcde".to_string();
    let input1 = "edbac".to_string();
    let expected = 12;
    assert_eq!(Solution::find_permutation_difference(input0, input1), expected);
}
